### PR TITLE
docs: make separate sphinx docs

### PR DESCRIPTION
**Pull request by jsstevenson**
I used postgres for the quick start instruction because it required fewer steps to write out (lazy) but I could rewrite it for dynamo. Hopefully this is a more ready/finished PR than the AnyVar FastAPI PR?

I have no idea what deployment looks like (I assume we put it all on readthedocs) but we could also add a build/deploy docs action

Ideally you could condense the README to refer to things in documentation as well but I didn't try it

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -59,6 +59,8 @@ dev =
     pytest-cov
     mock
     ipykernel
+    sphinx == 6.1.3
+    sphinx-autodoc-typehints == 1.22.0
```
Comment (jsstevenson) [inline setup.cfg:63]: I read somewhere that you should pin exact docs versions to ensure consistent builds moving forward, which I think makes sense

File: docs/source/contributing.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+This does overwrite existing data, so be sure to use a separate database instance from your main working environment.
+
+
+Documentation
+-------------
+
+The documentation is built with Sphinx, which is included as part of the developer dependencies. To build a local copy, navigate to the `docs/` subdirectory and use `make` to build the HTML version: ::
+
+    cd docs
+    make html
```
Comment (korikuzma) [inline docs/source/contributing.rst:68]:
```(gene-normalization) bash-3.2$ make html
Running Sphinx v6.1.3
WARNING: html_static_path entry '_static' does not exist
```

Setting `html_static_path = []` in conf.py removes this warning message.

**Review by korikuzma — COMMENTED** [review]
I think it would be helpful to do a quick start for dynamodb, but this could be done in a separate issue. 

I also assume that when we clean up our docstrings, the docs will update to show members, undoc-members, and show-inheritance?

**Comment by jsstevenson** [comment]
> I also assume that when we clean up our docstrings, the docs will update to show members, undoc-members, and show-inheritance?

@korikuzma Could you explain what you mean? Where relevant, the built docs should be showing all of this already (I don't think we have anything that would be caught by `undoc-members` but it seemed like a thing to leave in just in case we did someday).

**Comment by jsstevenson** [comment]
Some other outstanding questions to settle someday:

- theme?
- there's a fairly extensive way to get exact line numbers for source links. seems like overkill for now but stashing for reference https://github.com/sphinx-doc/sphinx/issues/1556#issuecomment-1010273175
- it'd be nice to list releases or a changelog or something. one way to do it automatically from commit messages: https://sphinx-git.readthedocs.io/en/latest/getting-started.html

**Comment by korikuzma** [comment]
> > I also assume that when we clean up our docstrings, the docs will update to show members, undoc-members, and show-inheritance?
> 
> @korikuzma Could you explain what you mean? Where relevant, the built docs should be showing all of this already (I don't think we have anything that would be caught by `undoc-members` but it seemed like a thing to leave in just in case we did someday).

Ya, so like [here](url). It looks pretty empty so I was wondering if when we remove the types in the docstrings if more things would appear

**Comment by korikuzma** [comment]
> Some other outstanding questions to settle someday:
> 
>     * theme?
> 
>     * there's a fairly extensive way to get exact line numbers for source links. seems like overkill for now but stashing for reference [autolinkcode extension sphinx-doc/sphinx#1556 (comment)](https://github.com/sphinx-doc/sphinx/issues/1556#issuecomment-1010273175)
> 
>     * it'd be nice to list releases or a changelog or something. one way to do it automatically from commit messages: https://sphinx-git.readthedocs.io/en/latest/getting-started.html

@jsstevenson Do you want to make issues for these?

File: docs/source/search_capabilities.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+The Gene Normalizer provides three different search modes:
+
+ * **search**: for each source, find the record or records that best match the given search string.
+ * **normalize**: find the normalized concept that best matches the given search string. Return a merged record that incorporates data from all associated records from each source. See :ref:`build_normalization` for more information.
+ * **normalize_unmerged**: find the normalized concept that best matches the given search string. Return each source record associated with that normalized concept.
+
+REST endpoints
+--------------
+
+Once :ref:`HTTP service is activated<starting-service>`, OpenAPI documentation for the REST endpoints is available at the ``/genes`` subdirectory (e.g., with default service parameters, at ``http://localhost:8000/genes``). This documentation describes endpoint parameters and response objects, and provides some minimal example queries.
```
Comment (korikuzma) [inline docs/source/search_capabilities.rst:16]:
```suggestion
Once :ref:`HTTP service is activated<starting-service>`, OpenAPI documentation for the REST endpoints is available at the ``/genes`` endpoint (e.g., with default service parameters, at ``http://localhost:8000/genes``). This documentation describes endpoint parameters and response objects, and provides some minimal example queries.
```

**Comment by jsstevenson** [comment]
> Ya, so like [here](https://github.com/cancervariants/gene-normalization/blob/new-docs/docs/source/api/etl_api.rst). It looks pretty empty so I was wondering if when we remove the types in the docstrings if more things would appear

OH I meant to ask you about this also. This is because those classes have very few non-private methods. We could also provide `:private-members:` to all classes to capture those as well.

File: gene/etl/ensembl.py [hunk 4]
```diff
@@ -168,23 +170,25 @@ def _add_attributes(self, f, gene):
...
         https://vr-spec.readthedocs.io/en/1.1/terms_and_model.html#sequencelocation
 
-        :param gffutils.feature.Feature f: A gene from the data
-        :param dict gene: A transformed gene record
-        :param dict accession_numbers: Accession numbers for each chromosome
-            and scaffold
-        :param  SeqRepo sr: Access to the seqrepo
+        :param f: A gene from the data
+        :param gene: A transformed gene record
+        :param SeqRepo sr: Access to the seqrepo
```
Comment (korikuzma) [inline gene/etl/ensembl.py:181]:
```suggestion
        :param sr: Access to the seqrepo
```

File: gene/etl/vrs_locations/chromosome_location.py [hunk 5]
```diff
@@ -14,14 +15,13 @@
 class ChromosomeLocation:
     """The class for GA4GH Chromosome Location."""
 
-    def get_location(self, location, gene):
+    def get_location(self, location: Dict, gene: Dict):
```
Comment (korikuzma) [inline gene/etl/vrs_locations/chromosome_location.py:18]:
```suggestion
    def get_location(self, location: Dict, gene: Dict) -> Optional[Dict]:
```

**Review by korikuzma — DISMISSED** [review]
@jsstevenson This is awesome!! 💯

**Review by korikuzma — APPROVED** [review]

**Review by korikuzma — APPROVED** [review]
💯

**Pull request merged**